### DISCUSSION TITLE
Displaying ChatGPT-Generated Charts in Dify

### DISCUSSION THREAD

**Discussion by yyipkei123** [langgenius/dify#10910] [Help] [answered]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I would like to know if Dify can display charts generated by ChatGPT-4 as a data analytics agent. While the LLM can generate the charts, it seems unable to display them directly in Dify. Are there any alternatives or tools within Dify that can achieve this functionality? Thanks!

**Comment by crazywoola** [comment, marked as answer]
This PR https://github.com/langgenius/dify/pull/6961 add this feature long time ago. You could use following syntax to render charts.
```echarts
{
  "xAxis": {
    "type": "category",
    "data": ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
  },
  "yAxis": {
    "type": "value"
  },
  "series": [
    {
      "data": [120, 200, 150, 80, 70, 110, 130],
      "type": "bar",
      "showBackground": true,
      "backgroundStyle": {
        "color": "rgba(180, 180, 180, 0.2)"
      }
    }
  ]
}
```

Which should render something into. 

![image](https://github.com/user-attachments/assets/6e9a8b6c-1ba9-40bd-931c-4c1db2a91b4e)
